File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/java/chartModifier2D/ChartViewModel.java
package com.scichart.docsandbox.examples.java.chartModifier2D;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

// <ChartModifierToViewModel>
public class ChartViewModel extends ViewModel {
    private final MutableLiveData<String> rolloverData = new MutableLiveData<>();

    public LiveData<String> getRolloverData() {
        return rolloverData;
    }

    // Call this method when you detect rollover changes
    public void onRolloverChanged(String seriesName, String xValue, String yValue) {
        String data = seriesName + ": X=" + xValue + ", Y=" + yValue;
        rolloverData.setValue(data);
    }
}
// </ChartModifierToViewModel>

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/chartModifier2D/ChartViewModel.kt
package com.scichart.docsandbox.examples.kotlin.chartModifier2D

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

// <ChartModifierToViewModel>
class ChartViewModel : ViewModel() {
    private val _rolloverData = MutableLiveData<String>()
    val rolloverData: LiveData<String> = _rolloverData

    // Call this method when you detect rollover changes
    fun onRolloverChanged(seriesName: String, xValue: String, yValue: String) {
        val data = "$seriesName: X=$xValue, Y=$yValue"
        _rolloverData.value = data
    }
}
// </ChartModifierToViewModel>